client/app/pages/home/Home.jsx

```diff
@@ -173,15 +173,12 @@ function Home() {
   );
 }
 
-export default function init(ngModule) {
-  ngModule.component("homePage", react2angular(Home));
-
-  return {
-    "/": {
-      template: "<home-page></home-page>",
-      title: "Redash",
-    },
-  };
-}
-
-init.init = true;
+export default {
+  path: "/",
+  title: "Redash",
+  render: (routeParams, currentRoute, location) => (
+    <AuthenticatedPageWrapper key={location.path}>
```

Comment:
They all should have `key` attribute - I need to check it :thinking: In short - `key` allows React to know if component instance of the same type is still the same (and should be just updated), or new one (to destroy old and initialize new). In this case I set `key` to current URL path because we use the same component for different pages (e.g. there are four query lists (all/my/favorites/archive), two dashboard lists, two query editors (new/existing query), etc. All that components wasn't designed to handle updates - they do some stuff on construction and then live their life. Therefore if you try to navigate between, say, query lists (without `key` attribute) - React will decide that that's the same component and try to update it; component ignores that and shows previous data.

First I was thinking about setting that `key` in Router, but later realized that it may be useful (and even needed sometimes): for example, we have Query Snippets pages (list and edit) which use the same controller; that controller can handle all updates, therefore for all routes it should have the same key (or not have the key at all).

Reply:
Upd.: I forgot to move `key` attributes for some controllers when adding wrappers :man_facepalming: